pkg/network/usm/ebpf_gotls.go

```diff
@@ -307,9 +308,28 @@ func registerCBCreator(mgr *manager.Manager, offsetsDataMap *ebpf.Map, probeIDs
 	}
 }
 
+// DetachPID detaches the provided PID from the eBPF program.
+func (p *goTLSProgram) DetachPID(pid uint32) error {
+	return p.registry.Unregister(pid)
+}
```

Comment:
I would add something like
```
var _ util.Attacher = &goTLSProgram{}
``` 

To the top of every protocol that implements the interface.
IMO that helps to convey to reader that these methods are part of an interface and also gives you a compilation-time check if the interface ever changes.